### ISSUE TITLE
 reflect.Value.Set using value obtained using unexported field

### ISSUE THREAD

**Issue by nesymno** [dranikpg/dto-mapper#5]
Hey! I'm trying to convert `uuid.UUID` to string, but getting error:
` reflect.Value.Set using value obtained using unexported field`

My conv func looks like this:
```golang
m := dto.Mapper{}
m.AddConvFunc(func(id uuid.UUID, m *dto.Mapper) (string, error) {
    return id.String(), nil
})
```

I try to map next struct:

```golang
// User is the model entity for the User schema.
type User struct {
	config `json:"-"`
	// ID of the ent.
	ID uuid.UUID `json:"id,omitempty"`
	// CreatedAt holds the value of the "created_at" field.
	CreatedAt time.Time `json:"created_at,omitempty"`
	// UpdatedAt holds the value of the "updated_at" field.
	UpdatedAt time.Time `json:"updated_at,omitempty"`
	// DeletedAt holds the value of the "deleted_at" field.
	DeletedAt *time.Time `json:"deleted_at,omitempty"`
	// AuthID holds the value of the "auth_id" field.
	AuthID uuid.UUID `json:"auth_id,omitempty"`
	// Email holds the value of the "email" field.
	Email string `json:"email,omitempty"`
	// Username holds the value of the "username" field.
	Username string `json:"username,omitempty"`
	// Password holds the value of the "password" field.
	Password string `json:"-"`
	// FirstName holds the value of the "first_name" field.
	FirstName string `json:"first_name,omitempty"`
	// LastName holds the value of the "last_name" field.
	LastName string `json:"last_name,omitempty"`
	// Bio holds the value of the "bio" field.
	Bio string `json:"bio,omitempty"`
	// Location holds the value of the "location" field.
	Location string `json:"location,omitempty"`
	// Github holds the value of the "github" field.
	Github string `json:"github,omitempty"`
	// Gitlab holds the value of the "gitlab" field.
	Gitlab string `json:"gitlab,omitempty"`
	// Microsoft holds the value of the "microsoft" field.
	Microsoft string `json:"microsoft,omitempty"`
	// Linkedin holds the value of the "linkedin" field.
	Linkedin string `json:"linkedin,omitempty"`
	// Google holds the value of the "google" field.
	Google string `json:"google,omitempty"`
	// DefOrgID holds the value of the "def_org_id" field.
	DefOrgID *uuid.UUID `json:"def_org_id,omitempty"`
	// Timezone holds the value of the "timezone" field.
	Timezone string `json:"timezone,omitempty"`
	// Flags holds the value of the "flags" field.
	Flags int `json:"flags,omitempty"`
	// Prefs holds the value of the "prefs" field.
	Prefs map[string]interface{} `json:"prefs,omitempty"`
	// LoginAt holds the value of the "login_at" field.
	LoginAt *time.Time `json:"login_at,omitempty"`
	// Edges holds the relations/edges for other nodes in the graph.
	// The values are being populated by the UserQuery when eager-loading is set.
	Edges        UserEdges `json:"edges"`
	selectValues sql.SelectValues
}
```

That is ent model and it is autogenerated, so I can't to edit it (I can, but after regenerate all updates will be lost).

Help pls!

**Comment by dranikpg** [comment]
Hi! Thanks for reporting the issue. I will fix it today.

In short, whenever conversion / inspection functions are defined, I skip direct assignability of structs out of fear that we might miss running a conversion function 

However I guess conversion functions are used only when the types really aren't assignable, so it would be okay to skip them 🤔

**Comment by dranikpg** [comment]
Fixed v0.2.1
